monai/apps/auto3dseg/ensemble_builder.py

```diff
@@ -582,9 +582,18 @@ def ensemble(self):
         builder.set_ensemble_method(self.ensemble_method)
         self.ensembler = builder.get_ensemble()
         infer_files = self.ensembler.infer_files
-        infer_files = partition_dataset(
-            data=infer_files, shuffle=False, num_partitions=self.world_size, even_divisible=True
-        )[self.rank]
+        if len(infer_files) < self.world_size:
+            if len(infer_files) == 0:
+                logger.info("No existing infer files. Ensembler ending.")
+                return
+            infer_files = partition_dataset(data=infer_files, shuffle=False,
+                                            num_partitions=len(infer_files))[self.rank] \
+                                            if self.rank < len(infer_files) else []
+        else:
+            infer_files = partition_dataset(
+                data=infer_files, shuffle=False, num_partitions=self.world_size, even_divisible=True
+            )[self.rank]
+
         # TO DO: Add some function in ensembler for infer_files update?
         self.ensembler.infer_files = infer_files
```

Comment:
do we need the logic below (including self.ensembler(pred_param=self.kwargs) ) when infer_files is empty list?

are you sure self.ensembler() will not crash with an empty list? may be it's saver to write, 

```
if len(infer_files) >0:
  self.ensembler(...)
  
```

Reply:
if infer_files is empty, it's returned. I will test if it will crash
```
        if len(infer_files) < self.world_size:
            if len(infer_files) == 0:
                logger.info(f"No existing infer files. Ensembler ending.")
                return 
```